pkg/bindings/connection.go

```diff
@@ -93,14 +96,34 @@ func NewConnection(ctx context.Context, uri string) (context.Context, error) {
 // or unix:///run/podman/podman.sock
 // or ssh://<user>@<host>[:port]/run/podman/podman.sock
 func NewConnectionWithIdentity(ctx context.Context, uri string, identity string, machine bool) (context.Context, error) {
-	var err error
-	if v, found := os.LookupEnv("CONTAINER_HOST"); found && uri == "" {
-		uri = v
-	}
+	return NewConnectionWithOptions(ctx, Options{URI: uri, Identity: identity, Machine: machine})
+}
 
-	if v, found := os.LookupEnv("CONTAINER_SSHKEY"); found && len(identity) == 0 {
-		identity = v
+type Options struct {
+	URI         string
+	Identity    string
+	TLSCertFile string
+	TLSKeyFile  string
+	TLSCAFile   string
+	Machine     bool
+}
+
+func orEnv(s string, env string) string {
+	if len(s) != 0 {
+		return s
 	}
+	s, _ = os.LookupEnv(env)
+	return s
+}
+
+func NewConnectionWithOptions(ctx context.Context, opts Options) (context.Context, error) {
+	var err error
+
+	uri := orEnv(opts.URI, "CONTAINER_HOST")
+	identity := orEnv(opts.Identity, "CONTAINER_SSHKEY")
+	tlsCertFile := orEnv(opts.TLSCertFile, "CONTAINER_TLS_CERT")
+	tlsKeyFile := orEnv(opts.TLSKeyFile, "CONTAINER_TLS_KEY")
+	tlsCAFile := orEnv(opts.TLSCAFile, "CONTAINER_TLS_CA")
```

Comment:
I think the env var lookup should be done in containers/common. There, we can set the values to be read first from the env var, then from the containers.conf value.

Reply:
Still applied. Should be plumbed through the CLI through common.